File: Folicon_Native/Classes/DArtBrowseResult.vb
    Imports Newtonsoft.Json

Public Class Author

        <JsonProperty("userid")>
        Public Property Userid As String

        <JsonProperty("username")>
        Public Property Username As String

        <JsonProperty("usericon")>
        Public Property Usericon As String

        <JsonProperty("type")>
        Public Property Type As String
    End Class

    Public Class Stats

        <JsonProperty("comments")>
        Public Property Comments As Integer

        <JsonProperty("favourites")>
        Public Property Favourites As Integer
    End Class

    Public Class Content

        <JsonProperty("src")>
        Public Property Src As String

        <JsonProperty("filesize")>
        Public Property Filesize As Integer

        <JsonProperty("height")>
        Public Property Height As Integer

        <JsonProperty("width")>
        Public Property Width As Integer

        <JsonProperty("transparency")>
        Public Property Transparency As Boolean
    End Class

    Public Class Thumb

        <JsonProperty("src")>
        Public Property Src As String

        <JsonProperty("height")>
        Public Property Height As Integer

        <JsonProperty("width")>
        Public Property Width As Integer

        <JsonProperty("transparency")>
        Public Property Transparency As Boolean
    End Class

    Public Class Result

        <JsonProperty("deviationid")>
        Public Property Deviationid As String

        <JsonProperty("printid")>
        Public Property Printid As String

        <JsonProperty("url")>
        Public Property Url As String

        <JsonProperty("title")>
        Public Property Title As String

        <JsonProperty("category")>
        Public Property Category As String

        <JsonProperty("category_path")>
        Public Property CategoryPath As String

        <JsonProperty("is_downloadable")>
        Public Property IsDownloadable As Boolean

        <JsonProperty("is_mature")>
        Public Property IsMature As Boolean

        <JsonProperty("is_favourited")>
        Public Property IsFavourited As Boolean

        <JsonProperty("is_deleted")>
        Public Property IsDeleted As Boolean

        <JsonProperty("author")>
        Public Property Author As Author

        <JsonProperty("stats")>
        Public Property Stats As Stats

        <JsonProperty("published_time")>
        Public Property PublishedTime As Integer

        <JsonProperty("allows_comments")>
        Public Property AllowsComments As Boolean

        <JsonProperty("content")>
        Public Property Content As Content

        <JsonProperty("thumbs")>
        Public Property Thumbs As Thumb()
    End Class

    Public Class DArtBrowseResult

        <JsonProperty("has_more")>
        Public Property HasMore As Boolean

        <JsonProperty("next_offset")>
        Public Property NextOffset As Integer

        <JsonProperty("results")>
        Public Property Results As Result()
    End Class

File: FoliCon/Models/Api/DArtBrowseResult.cs
// ReSharper disable StringLiteralTypo
// ReSharper disable IdentifierTypo
namespace FoliCon.Models.Api;

public class Author
{
    [JsonProperty("userid")]
    public string Userid { get; set; }

    [JsonProperty("username")]
    public string Username { get; set; }

    [JsonProperty("usericon")]
    public string Usericon { get; set; }

    [JsonProperty("type")]
    public string Type { get; set; }
}

public class PremiumFolderData
{
    [JsonProperty("type")]
    public string Type { get; set; }
    
    [JsonProperty("has_access")]
    public bool HasAccess { get; set; }
    
    [JsonProperty("gallery_id")]
    public string GallaryId { get; set; }
}

public class Stats
{
    [JsonProperty("comments")]
    public int Comments { get; set; }

    [JsonProperty("favourites")]
    public int Favourites { get; set; }
}

public class Content
{
    [JsonProperty("src")]
    public string Src { get; set; }

    [JsonProperty("filesize")]
    public int Filesize { get; set; }

    [JsonProperty("height")]
    public int Height { get; set; }

    [JsonProperty("width")]
    public int Width { get; set; }

    [JsonProperty("transparency")]
    public bool Transparency { get; set; }
}

public class Thumb
{
    [JsonProperty("src")]
    public string Src { get; set; }

    [JsonProperty("height")]
    public int Height { get; set; }

    [JsonProperty("width")]
    public int Width { get; set; }

    [JsonProperty("transparency")]
    public bool Transparency { get; set; }
}

public class Result
{
    [JsonProperty("deviationid")]
    public string Deviationid { get; set; }

    [JsonProperty("printid")]
    public string Printid { get; set; }

    [JsonProperty("url")]
    public string Url { get; set; }

    [JsonProperty("title")]
    public string Title { get; set; }

    [JsonProperty("category")]
    public string Category { get; set; }

    [JsonProperty("category_path")]
    public string CategoryPath { get; set; }

    [JsonProperty("is_downloadable")]
    public bool IsDownloadable { get; set; }

    [JsonProperty("is_mature")]
    public bool IsMature { get; set; }

    [JsonProperty("is_favourited")]
    public bool IsFavourited { get; set; }

    [JsonProperty("is_deleted")]
    public bool IsDeleted { get; set; }

    [JsonProperty("author")]
    public Author Author { get; set; }

    [JsonProperty("stats")]
    public Stats Stats { get; set; }

    [JsonProperty("published_time")]
    public int PublishedTime { get; set; }

    [JsonProperty("allows_comments")]
    public bool AllowsComments { get; set; }

    [JsonProperty("content")]
    public Content Content { get; set; }

    [JsonProperty("thumbs")]
    public Thumb[] Thumbs { get; set; }
    
    [JsonProperty("premium_folder_data")]
    public PremiumFolderData PremiumFolderData { get; set; }
}

public class DArtBrowseResult
{
    [JsonProperty("has_more")]
    public bool HasMore { get; set; }

    [JsonProperty("next_offset")]
    public int NextOffset { get; set; }

#nullable enable
    [JsonProperty("results")]
    public Result[]? Results { get; set; }
}